File: backend/src/models/collateral.model.ts
import {Schema, model } from 'mongoose';
import { CollateralStatus,DocumentType, ICollateral } from '../interfaces/collateral.interface';

const CollateralSchema = new Schema<ICollateral>(
    {
        userId: { type: Schema.Types.ObjectId, ref: "User"},
        documentType: { 
            type: String,
            enum: Object.values(DocumentType),
            required:true},
        file: { type: String, required:true },
        adminRemark: { type: String },
        status: { 
            type: String,
            enum: Object.values(CollateralStatus),
            default: CollateralStatus.PENDING }
    },
    { timestamps: true } 
);

export default model<ICollateral>("Collateral", CollateralSchema);

File: frontend/lib/models/collateral_model.dart
class Collateral {
  final String id;
  final String file;
  final String documentType;
  final String status;

  Collateral({
    required this.id,
    required this.file,
    required this.documentType,
    required this.status,
  });

factory Collateral.fromJson(Map<String, dynamic> json) {
    return Collateral(
      id: json["_id"],
      file: json["file"],

      documentType: json["documentType"],
      status: json["status"],
    );
  }
}